src/index.js

```diff
@@ -1,62 +1,137 @@
+
+  list.innerHTML = listsCode.join('');
+
+  const checkbox = document.querySelectorAll('.check-box');
+  checkBox(checkbox);
 
-  const btn = document.createElement('li');
-  btn.innerHTML = '<button type="button">Clear all completed</button>';
-  list.append(btn);
+  const btn = document.querySelectorAll('.icon-d');
+  btn.forEach((el) => {
+    el.addEventListener('click', (e) => {
+      const oldList = e.target.parentElement.parentElement.getAttribute('data-id'); // get parent
+      ToDoClass.remove(oldList);
+      showList();
+    });
+  });
 }
 
 showList();
 
 document.addEventListener('DOMContentLoaded', () => {
-  if (toDoList[0].completed === true) {
-    document.getElementById('0').classList.add('strike');
-    document.querySelector('.list-0').checked = true;
-  }
+  const submitBtn = document.querySelector('.sub-icon');
+  submitBtn.addEventListener('click', () => {
+    const desInput = document.querySelector('.text');
+    const description = desInput.value.trim();
+    if (!description) {
+      return;
+    }
 
-  if (toDoList[1].completed === true) {
-    document.getElementById('1').classList.add('strike');
-    document.querySelector('.list-1').checked = true;
-  }
+    let id = 0;
+    if (toDoList.length === 0) {
+      id = toDoList.length + 1;
+    } else if (toDoList.length > 0) {
+      id = toDoList[toDoList.length - 1].index + 1;
+    }
 
-  const checkbox = document.querySelectorAll('.check-box');
-  checkBox(checkbox);
-});
+    const index = id;
+
+    const completed = false;
+    const newList = new ToDoClass(description, completed, index);
+    ToDoClass.addTo(newList);
+    showList();
+    desInput.value = '';
+  });
+
+  const clearCompleted = document.querySelector('.clear-all');
+  clearCompleted.addEventListener('click', () => {
+    toDoList = toDoList.filter((x) => x.completed === false);
+    showList();
+    for (let i = 0; i < toDoList.length; i += 1) {
+      toDoList[i].index = i + 1;
+    }
+    localStorage.setItem('ToDo', JSON.stringify(toDoList));
+  });
+
+  const isComplete = document.querySelectorAll('.check-box');
+  isComplete.forEach((el) => {
+    for (let i = 0; i < toDoList.length; i += 1) {
+      if (toDoList[i].completed === true) {
+        const compId = el.parentElement.getAttribute('id');
+        if (Number(compId) === toDoList[i].index) {
+          el.parentElement.classList.add('strike');
+          el.checked = true;
+        }
+      }
+    }
+  });
+
+  const saveBtn = document.querySelector('.save');
+  const paragraphs = document.querySelectorAll('.edit');
+  paragraphs.forEach((el) => {
+    el.addEventListener('click', () => {
+      saveBtn.addEventListener('click', () => {
+        const newInput = el.innerHTML;
+        for (let i = 0; i < toDoList.length; i += 1) {
+          const parentId = el.parentElement.getAttribute('id');
+          if (Number(parentId) === toDoList[i].index) {
+            toDoList[i].description = newInput;
+          }
```

Comment:
- [ ] OPTIONAL - Kindly ensure that before you merge the PR you fix the issue you mentioned in the comment as regards the edited todo item not being properly saved.